### ISSUE TITLE
simplify remote plugins

### ISSUE THREAD

**Issue by justinmk** [neovim/neovim-ruby#108]
# Problem

The design, and usage, of remote plugins can be simplified, as described in https://github.com/neovim/neovim/issues/27949 

tl;dr:

- remove the concept of "remote plugins"
- instead, any Ruby module that imports the neovim-ruby client can call [register_handler](https://github.com/neovim/neovim-ruby/pull/107/files#diff-c9658e11bddd68483313e383bfb3d7c3c9e0f544e98a057c432de149b4066c43) to handle requests from Nvim
- "remote plugins" become plain old Lua plugins which call RPC methods on the Go module

# Solution

- [ ] Apply the above proposal to neovim-ruby client.
    - [ ] "Encapsulate the stdio related code in a helper function." [Neovim.start_remote()](https://github.com/neovim/neovim-ruby/pull/107/files#diff-c9658e11bddd68483313e383bfb3d7c3c9e0f544e98a057c432de149b4066c43)
    - [x] Update docs https://github.com/neovim/neovim-ruby/pull/107
- [ ] Deprecate the old rplugin interface.
    - https://github.com/neovim/go-client/pull/173
- [ ] Remove old rplugin-related code

- related pynvim issue: https://github.com/neovim/pynvim/issues/567

**Comment by alexgenco** [comment]
>Update handling of g:ruby_host_prog, so it can point to ruby

>(Nvim 0.12) Remove old rplugin-related code

Is there anything to do for these in this repo? These seem like changes to neovim itself, except maybe deleting the remote plugin code once nvim 0.12 lands.

**Comment by justinmk** [comment]
Correct. Just for tracking/visibility
